delegator.tf

```diff
@@ -0,0 +1,21 @@
+# Service account to delegate permission for DNS zone via IAM
+resource "google_service_account" "delegator" {
+  account_id   = local.resource_name
+  display_name = "${local.resource_name} service account"
+}
+
+resource "google_project_iam_custom_role" "dns-delegator" {
+  role_id     = replace("dns_delegator_${local.resource_name}", "-", "_")
+  title       = "DNS delegator ${local.resource_name}"
+  description = "DNS delegator for ${local.resource_name}"
+  permissions = ["dns.changes.create", "dns.resourceRecordSets.update", "dns.resourceRecordSets.delete"]
```

Comment:
Can we scope these to the dns zone we're creating in this module?

Reply:
GCP provides IAM conditions to allow permission to specific resources. As per their doc, there isn't support to filter DNS resource https://cloud.google.com/iam/docs/conditions-resource-attributes#resource-name.

Reply:
Bummer.